### PR TITLE
allow prefix in oranda serve

### LINKED ISSUE #71: Add oranda theme dev
Add command to build & serve oranda with a link to a theme config that includes all possible items (repo bar, multiple pages, README with markdown elements). Predicated on https://github.com/axodotdev/oranda/issues/70

### LINKED ISSUE #90: oranda serve does not play nicely with `pathprefix`
pages

### PR DESCRIPTION
**Pull request by SaraVieira**
closes https://github.com/axodotdev/oranda/issues/90

closes https://github.com/axodotdev/oranda/issues/71

### CODE REVIEW

File: src/site/header.rs [hunk 1]
```diff
@@ -41,7 +41,7 @@ pub fn create(config: &Config) -> Result<Box<header<String>>> {
                 let mut href = format!("/{}", file_name);
 
                 if let Some(prefix) = &config.path_prefix {
-                    href = format!("/{}{}", prefix, href);
```
Comment (SaraVieira) [inline src/site/header.rs:44]: This seems to be a fix but it's super cumbersome

https://github.com/tokio-rs/axum/discussions/446

Comment (ashleygwilliams) [reply src/site/header.rs:44]: oh interesting, def a bummer. might be a nice opportunity to contribute back. regardless maybe we can also snag some of the error work from that answer so we don't just serve the empty page if there's an issue?

File: src/commands/serve.rs [hunk 2]
```diff
@@ -33,10 +33,20 @@ impl Serve {
             },
         );
 
-        let app = Router::new().fallback(static_files_service);
+        let app;
+        let route;
+        if let Some(path_prefix) = config.path_prefix {
+            route = format!("/{}", path_prefix);
```
Comment (ashleygwilliams) [inline src/commands/serve.rs]: is the slash necessary? my instinct is it wouldn't be but i'm not sure

Comment (ashleygwilliams) [reply src/commands/serve.rs]: i think this is a scenario where we may want to use `Url` https://docs.rs/url/latest/url/struct.Url.html#method.join

**Review by ashleygwilliams — APPROVED** [review]
small requests- if you don't do the error handling in this one we should open an issue for it

File: src/commands/serve.rs (hunk 2 above)
Comment (SaraVieira) [reply src/commands/serve.rs]: oh no

```
thread 'main' panicked at 'Paths must start with a `/`', src/commands/serve.rs:41:18
```

Comment (ashleygwilliams) [reply src/commands/serve.rs]: well that settles that ha

File: src/site/header.rs (hunk 1 above)
Comment (SaraVieira) [reply src/site/header.rs:44]: We still need this right?

Comment (ashleygwilliams) [reply src/site/header.rs:44]: no i dont think so, i ran it locally and it worked?

**Comment by SaraVieira** [comment]
Can't approve because it's my PR but ship it!

**Pull request merged**